app/controllers/application_controller/ci_processing.rb

```diff
@@ -215,67 +215,74 @@ def breadcrumb_name(_model)
     ui_lookup(:models => self.class.model.name)
   end
 
+  def check_retire_requirements(selected_items)
+    # FIXME: should check model, not controller
+    return true if %w(orchestration_stack service).include?(controller_name)
+
+    if VmOrTemplate.find(selected_items).any? { |vm| !vm.supports_retire? }
+      add_flash(_("Retire does not apply to selected %{model}") %
+        {:model => ui_lookup(:table => "miq_template")}, :error)
```

Comment:
The `:model` here should be expanded to the full string, for the sake of translators / translations.